digiwf-integrations/digiwf-s3-integration/digiwf-s3-integration-core/src/main/java/io/muenchendigital/digiwf/s3/integration/adapter/in/rest/FileController.java

```diff
@@ -0,0 +1,122 @@
+package io.muenchendigital.digiwf.s3.integration.adapter.in.rest;
+
+import io.muenchendigital.digiwf.s3.integration.adapter.in.rest.dto.FileDataDto;
+import io.muenchendigital.digiwf.s3.integration.adapter.in.rest.dto.PresignedUrlDto;
+import io.muenchendigital.digiwf.s3.integration.adapter.in.rest.mapper.FileDataMapper;
+import io.muenchendigital.digiwf.s3.integration.adapter.in.rest.mapper.PresignedUrlMapper;
+import io.muenchendigital.digiwf.s3.integration.adapter.in.rest.validation.FolderInFilePath;
+import io.muenchendigital.digiwf.s3.integration.application.port.in.FileExistenceException;
+import io.muenchendigital.digiwf.s3.integration.application.port.in.FileOperationsInPort;
+import io.muenchendigital.digiwf.s3.integration.application.port.in.FileSystemAccessException;
+import io.muenchendigital.digiwf.s3.integration.domain.model.FileData;
+import io.muenchendigital.digiwf.s3.integration.domain.model.PresignedUrl;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.time.LocalDate;
+
+@Slf4j
+@Validated
+@RestController
+@RequiredArgsConstructor
+@Tag(name = "FileAPI", description = "API to interact with files")
+@RequestMapping("/file")
+public class FileController {
+
+  private final FileOperationsInPort fileOperations;
+  private final FileDataMapper fileMapper;
+  private final PresignedUrlMapper presignedUrlMapper;
+
+  @GetMapping
+  @Operation(description = "Creates a presigned URL to fetch the file specified in the parameter from the S3 storage")
+  public ResponseEntity<PresignedUrlDto> get(@RequestParam @NotEmpty @Size(max = FileData.LENGTH_PATH_TO_FILE) @FolderInFilePath final String pathToFile,
+                                             @RequestParam @NotNull @Min(FileData.MIN_EXPIRES_IN_MINUTES) final Integer expiresInMinutes) {
+    try {
+      log.info("Received a request for S3 presigned url to download a file");
+      final PresignedUrl fileResponse = this.fileOperations.getFile(pathToFile, expiresInMinutes);
+      final PresignedUrlDto presignedUrlDto = this.presignedUrlMapper.model2Dto(fileResponse);
+      return ResponseEntity.ok(presignedUrlDto);
+    } catch (final FileSystemAccessException exception) {
```

Comment:
What is your opinion about Exception Handler via Controller Advices?

Reply:
We don't have a uniform mapping here. Therefor it will remain in controller.